include/grpc++/impl/codegen/completion_queue.h

```diff
@@ -31,8 +31,18 @@
  *
  */
 
-/// A completion queue implements a concurrent producer-consumer queue, with two
-/// main methods, \a Next and \a AsyncNext.
+/// A completion queue implements a concurrent producer-consumer queue, with
+/// two main API-exposed methods: \a Next and \a AsyncNext. These
+/// methods are the essential component of the gRPC C++ asynchronous API.
+/// There is also a Shutdown method to indicate that a given completion queue
+/// will no longer have regular events. This must be called before the
+/// completion queue is destroyed.
+/// All completion queue APIs are thread-safe and may be used concurrently with
+/// any other completion queue API invocation; it is acceptable to have
+/// multiple threads calling Next or AsyncNext on the same or different
+/// completion queues, or to call these methods concurrently with a Shutdown
+/// elsewhere. All of these should be completed, though, before a completion
```

Comment:
maybe worth separating the last important sentece about completion-before-destruction in a \remark paragraph (https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdremark)